src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java

```diff
@@ -72,13 +72,13 @@ public static void registerProvider(final IStandardRequestManager manager, final
 
         manager.getProviderBiMap().put(provider.getToken(), provider);
 
-        final ImmutableList.Builder<IToken> resolverListBuilder = new ImmutableList.Builder<>();
+        final ImmutableList.Builder<IToken<?>> resolverListBuilder = new ImmutableList.Builder<>();
         resolverListBuilder.addAll(ResolverHandler.registerResolvers(manager, provider.getResolvers()));
 
         manager.getProviderResolverMap().put(provider.getToken(), resolverListBuilder.build());
     }
 
-    public static void removeProvider(final IStandardRequestManager manager, final IToken token) throws IllegalArgumentException
+    public static void removeProvider(final IStandardRequestManager manager, final IToken<?> token) throws IllegalArgumentException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)